### DISCUSSION TITLE
Shorten link cannot be copied in iPhone

### DISCUSSION THREAD

**Discussion by TeenTin** [SinTan1729/chhoto-url#33] [Q&A]
The shorten links created cannot be copied in iPhone, no matter how hard I tried.  Would anyone please help?

**Comment by SinTan1729** [comment]
Are you accessing the url through http or https? If it's http, automatic copying won't work. You'll have to long press on the entries to copy the link.

If this doesn't work, please explain what exactly you're doing to copy the links. I don't use an iPhone, so can't confirm if there are any actual issues.

**Reply by TeenTin** [reply]
Thank you very much for your reply.  Would you please take a look at the following screen record ?
https://photos.app.goo.gl/seUEwVS3Shw8RuS66

**Reply by SinTan1729** [reply]
The copy uses https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText which is listed as supported on Safari. What version of Safari and iOS are you using?

**Reply by TeenTin** [reply]
My iOS is 18.0.1 and my Safari is the latest one.  Should be iOS problem because both Safari and Chrome are not working.

**Reply by TeenTin** [reply]
As a workaround, would you please create a link for "long pressing to copy"?

**Reply by SinTan1729** [reply]
This seems like a bug. I'll have to try testing with an iOS device if possible. For now, you can just add the shorturl to the ends of your url for it to work. In the example in your screen recording, `https://myurl.teentin.freeddns.org/dnfo` is the address that would be copied if it worked properly. And indeed, it works.

**Reply by TeenTin** [reply]
Thank you very much for your care.  I love this software very much.  Lite, Simple and Easy.  I hope this can be fixed soon.  Anyway, I think adding an extra link for "long pressing to copy" is good.

**Reply by SinTan1729** [reply]
Thank you for the kind words. I'll try to get a fix.

**Reply by TeenTin** [reply]
> Thank you for the kind words. I'll try to get a fix.

Would you consider modifying the GREEN message, in my case, from ...
"Short URL dnfo was copied to clipboard!"
to ...
"Short URL (myurl.teentin.freeddns.org/dnfo) was copied to clipboard!" ?

Would this not only make it a little clearer, but also make it possible to highlight and copy the link?

Thanks for your attention.

**Comment by SinTan1729** [comment]
Further discussions are in #36.
